include/SDL3/SDL_process.h

```diff
@@ -0,0 +1,271 @@
+ */
+extern SDL_DECLSPEC SDL_Process *SDLCALL SDL_CreateProcessWithProperties(SDL_PropertiesID props);
+
+#define SDL_PROP_PROCESS_CREATE_ARGS_POINTER            "SDL.process.create.args"
+#define SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER     "SDL.process.create.environment"
+#define SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN           "SDL.process.create.stdin"
+#define SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN          "SDL.process.create.stdout"
+#define SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN          "SDL.process.create.stderr"
+#define SDL_PROP_PROCESS_CREATE_REDIRECT_STDERR_BOOLEAN "SDL.process.create.redirect_stderr"
+
+/**
+ * Get the properties associated with a process.
+ *
+ * The following read-only properties are provided by SDL:
+ *
+ * - `SDL_PROP_PROCESS_PID_NUMBER`: the process ID of the process.
+ * - `SDL_PROP_PROCESS_STDIN_POINTER`: an SDL_IOStream that can be used to write input to the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN` set to true.
+ * - `SDL_PROP_PROCESS_STDOUT_POINTER`: a non-blocking SDL_IOStream that can be used to read output from the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN` set to true.
+ * - `SDL_PROP_PROCESS_STDERR_POINTER`: a non-blocking SDL_IOStream that can be used to read error output from the process, if it was created with `SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN` set to true.
+ *
+ * \param process the process to query.
+ * \returns a valid property ID on success or 0 on failure; call
+ *          SDL_GetError() for more information.
+ *
+ * \sa SDL_CreateProcess
+ * \sa SDL_CreateProcessWithProperties
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern SDL_DECLSPEC SDL_PropertiesID SDL_GetProcessProperties(SDL_Process *process);
+
+#define SDL_PROP_PROCESS_PID_NUMBER     "SDL.process.pid"
+#define SDL_PROP_PROCESS_STDIN_POINTER  "SDL.process.stdin"
+#define SDL_PROP_PROCESS_STDOUT_POINTER "SDL.process.stdout"
+#define SDL_PROP_PROCESS_STDERR_POINTER "SDL.process.stderr"
+
+/**
+ * Read all the output from a process.
+ *
+ * If a process was created with I/O enabled, you can use this function to read the output. This function blocks until the process is complete, capturing all output, and providing the process exit code.
+ *
+ * This is just a convenience function. If you need more control over the process, you can get the output stream from the process properties and read it directly.
+ *
+ * The data is allocated with a zero byte at the end (null terminated) for
+ * convenience. This extra byte is not included in the value reported via
+ * `datasize`.
+ *
+ * The data should be freed with SDL_free().
+ *
+ * \param process The process to read.
+ * \param datasize a pointer filled in with the number of bytes read, may be NULL.
+ * \param exitcode a pointer filled in with the process exit code if the process has exited, may be NULL.
```

Comment:
We should be clear about how the exit status is encoded, because there are several different conventions. I had to read the implementation to find out that "killed by signal 15" comes out as exit status 15, which is definitely wrong.

I would recommend Python's convention: exit statuses are non-negative numbers, and being killed by a signal is a negative number (e.g. being killed by `SIGTERM`, signal 15, comes out as `-15`). That would be something like:

```suggestion
 * \param exitcode a pointer filled in with a non-negative exit status, or a negative number to indicate that the process was killed by signal `-*exitcode`. May be NULL.
```